### PR TITLE
[sqllab]More granular sqllab logging

### PR DESCRIPTION
**Pull request by timifasubaa**
This PR improves how sqllab logs its performance by logging at a more granular level primarily for asynchronous queries. 

It logs the 
(for asynchroous queries) time spent pending
time spent in the database
time spent writing to results_backend
time spent time spent reading from results_backend

This PR depends on https://github.com/apache/incubator-superset/pull/5844

@mistercrunch @john-bodley

### CODE REVIEW

File: superset/migrations/versions/9a3c5cebcc67_.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# revision identifiers, used by Alembic.
+revision = '9a3c5cebcc67'
+down_revision = '6400c588de5e'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.alter_column('query', 'start_running_time', nullable=False, new_column_name='time_at_db')
```
Comment (mistercrunch) [inline superset/migrations/versions/9a3c5cebcc67_.py]: This table tends to have hanging locks and is hard to alter. I've had issue every time I've had to alter this model where it just completely lock up everything. We'd all need to schedule downtime to allow for this.

Comment (mistercrunch) [reply superset/migrations/versions/9a3c5cebcc67_.py]: Maybe we can use some sort of tracing instead.

Comment (mistercrunch) [reply superset/migrations/versions/9a3c5cebcc67_.py]: I mean just using `statsd.timing(DURATION)`

File: superset/sql_lab.py [hunk 2]
```diff
@@ -172,11 +174,13 @@ def handle_error(msg):
         cursor = conn.cursor()
         logging.info('Running query: \n{}'.format(executed_sql))
         logging.info(query.executed_sql)
+        query_start_time = utils.now_as_float()
         db_engine_spec.execute(cursor, query.executed_sql, async_=True)
         logging.info('Handling cursor')
         db_engine_spec.handle_cursor(cursor, query, session)
         logging.info('Fetching data: {}'.format(query.to_dict()))
         data = db_engine_spec.fetch_data(cursor, query.limit)
+        stats_logger.timing('time_in_database', utils.now_as_float() - query_start_time)
```
Comment (john-bodley) [inline superset/sql_lab.py]: I wonder if there's a better (more descriptive) name than `time_in_database`? May `time_executing_query`?

**Review by john-bodley — COMMENTED** [review]
For the `statsD` key it seems like they should potentially by more descriptive, i.e., start with the `sqllab` prefix so rather than `time_pending` it could be something like `sqllab.query.time_pending`.  You can look at Datadog for examples of how other applications define their keys.

File: superset/sql_lab.py (hunk 2 above)
Comment (timifasubaa) [reply superset/sql_lab.py]: DOne

**Comment by timifasubaa** [comment]
That makes sense. I updated the names.

**Review by john-bodley — APPROVED** [review]

**Pull request merged**